### PR TITLE
feat: Add docker compose file for a containerized development environment

### LINKED ISSUE #49: Dockerize this repo!
#### Issue Summary

It is really nice to be able to develop locally inside a docker container!  Please dockerize this repo! 
- Please setup docker-compose to use a local volume for the library files
- Please add information to the README and USAGE files about how to startup the container

### PR DESCRIPTION
**Pull request by michaeljdennis**
Fixes #49 

### Checklist
- [x] I have made a material change to the repo (functionality, testing, spelling, grammar)
- [x] I have read the [Contribution Guide] and my PR follows them.
- [x] I updated my branch with the master branch.
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added necessary documentation about the functionality in the appropriate .md file
- [x] I have added in line documentation to the code I modified

### Short description of what this PR does:
This PR adds a docker compose file for a containerized development environment.

### CODE REVIEW

**Comment by thinkingserious** [comment]
Hello @michaeljdennis,

When I try to run this, I get a

```bash
Recoverable fatal error: Object of class SendGrid\Client could not be converted to string in /php-http-client/examples/example.php on line 19
```

when I click on the `example.php` link from `localhost:8080`. 

When you clicked on that link did you receive the right response?

With Best Regards,

Elmer

**Comment by michaeljdennis** [comment]
Hi @thinkingserious. I will take a look and get back to you.

**Comment by thinkingserious** [comment]
Awesome, thanks @michaeljdennis!

**Comment by michaeljdennis** [comment]
Hi @thinkingserious,

I looked into this a bit and found that some methods had been lost in a merge. I have re-added them in the commits above. Also, it appears that the master branch has errors that are similar to the one you reported when running the example file. All unit tests are passing though, so we may need to look into those. I have included fixes that allow the example file to be run successfully. Please let me know if it works for you now.

**Comment by thinkingserious** [comment]
Awesome @michaeljdennis!

I've updated the status of this issue in our backlog.

File: lib/Client.php [hunk 1]
```diff
@@ -370,26 +370,33 @@ private function buildUrl($queryParams = null)
...
     {
-        $options = [
+        $options = array_replace(
+            [
                 CURLOPT_RETURNTRANSFER => true,
                 CURLOPT_HEADER => true,
                 CURLOPT_CUSTOMREQUEST => strtoupper($method),
                 CURLOPT_SSL_VERIFYPEER => $this->verifySSLCerts,
                 CURLOPT_FAILONERROR => false,
             ] + $this->curlOptions;
```
Comment (thinkingserious) [inline lib/Client.php:380]: Missing closing parenthesis.

**Review by thinkingserious — COMMENTED** [review]
Thank you for the PR!

When I try to run this, I get the following error:

```
int(403) string(56) "{"errors":[{"field":null,"message":"access forbidden"}]}"

int(403) string(56) "{"errors":[{"field":null,"message":"access forbidden"}]}"

int(403) string(56) "{"errors":[{"field":null,"message":"access forbidden"}]}"

int(403) string(56) "{"errors":[{"field":null,"message":"access forbidden"}]}"

int(403) string(56) "{"errors":[{"field":null,"message":"access forbidden"}]}"

int(403) string(56) "{"errors":[{"field":null,"message":"access forbidden"}]}"

int(403) string(56) "{"errors":[{"field":null,"message":"access forbidden"}]}"
```

**Comment by thinkingserious** [comment]
Closing until PR feedback is addressed.

**Pull request closed without merging**